text/0000-remove-windows-containers-support.md

```diff
@@ -0,0 +1,97 @@
+# Meta
+[meta]: #meta
+- Name: Remove Windows Containers Support
+- Start Date: 2024-04-11
+- Author(s): aidan.delaney
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+
+Retain Windows Containers support in the buildpacks specification, but remove Windows Containers support from `lifecycle`, `pack` CLI, and documentation.
+
+# Definitions
+[definitions]: #definitions
+
+Make a list of the definitions that may be useful for those reviewing. Include phrases and words that buildpack authors or other interested parties may not be familiar with.
+
+* [Windows Containers](https://learn.microsoft.com/en-us/virtualization/windowscontainers/about/): Containers derived from Windows Server Containers.  For example: `mcr.microsoft.com/dotnet/framework/sdk:4.8-windowsservercore-ltsc2022`
+
+# Motivation
+[motivation]: #motivation
+
+- There are no known users of Windows Container support within the buildpacks ecosystem.
+- Supporting Windows Containers is straightforward at a specification level.  However, the CNB project has neither the time nor the expertise to continue to support Windows Containers in our tooling.
+- Removing Windows Container support from CNB tooling will speed up development, simplify testing and impact no existing users.
+- Establish platform parity between `pack` and `kpack`.  `kpack` has never supported building Windows Containers.
+
+# What it is
+[what-it-is]: #what-it-is
+
+- We want to remove Windows Containers related documentation, but clearly state that the specification is platform neutral.
+- Remove Windows Containers support from `pack`, but retain the Windows platform release of the `pack` CLI.
+    * this ensures `pack` users can still invoke `pack` on a Command or Power Shell console
+- Remove Windows Containers support from `lifecycle`.  Lifecycle can remove the Windows platform release as it will then be guaranteed to run on a Linux-based container.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+
+# Migration
+[migration]: #migration
+
+Teams that require Windows Container support will have to
+
+* continue to use the last release of `pack` that supports Windows Containers,
```

Comment:
I have a question here: When not using the `creator`,  `pack` tries to download the latest `lifecycle` and then calculate the API version supported. 

This is an screen-shoot from [lifecycle](https://hub.docker.com/r/buildpacksio/lifecycle/tags) dockerhub 

![Screenshot from 2024-12-19 08-39-06](https://github.com/user-attachments/assets/05df3116-1d03-4ac6-82d8-96e8b492d1c4)

Once lifecycle stops supporting Windows, we will not have an image for `windows/amd64`, let's suppose this happens on lifecycle `>= 0.21.x` version. On the other hand, we will keep releasing pack binaries for windows. 

When a `pack` user on Windows runs a build I think we will need to handle the cases where the lifecycle image for  `windows/amd64` is not available and print a warning message instead of a error which I think is gonna be the case.